### PR TITLE
fix: replace regex usage with simple search

### PR DESCRIPTION
**Pull request by strehle**
Refactor use of regex to create a count statement
Issue found by Sonar

### CODE REVIEW

File: server/src/main/java/org/cloudfoundry/identity/uaa/resources/jdbc/JdbcPagingList.java [hunk 1]
```diff
@@ -92,20 +94,23 @@ public E get(int index) {
...
             throw new IndexOutOfBoundsException("The indexes provided are outside the bounds of this list.");
         }
-        return new SafeIteratorList<E>(super.subList(fromIndex, toIndex));
+        return new SafeIteratorList<>(super.subList(fromIndex, toIndex));
     }
 
     private String getCountSql(String sql) {
-        String result = sql.replaceAll("(?i)select (.*?) from (.*)", "select count(*) from $2");
-        int orderByPos = result.toLowerCase().lastIndexOf("order by");
+        String result = UaaStringUtils.getValidatedString(sql).toLowerCase();
```
Comment (coolgang123) [inline server/src/main/java/org/cloudfoundry/identity/uaa/resources/jdbc/JdbcPagingList.java]: Converting it to lower case will not work as intended. Consider the sql query with a where clause with upper case criteria:

SELECT Foo.id, FOO.NAME from foo where name = 'Bar'

File: server/src/test/java/org/cloudfoundry/identity/uaa/resources/jdbc/JdbcPagingListTests.java [hunk 2]
```diff
@@ -105,6 +106,23 @@ void jumpOverPages() {
         assertNotNull(map.get("name"));
     }
 
+    @Test
+    void selectColumnsFull() {
```
Comment (coolgang123) [inline server/src/test/java/org/cloudfoundry/identity/uaa/resources/jdbc/JdbcPagingListTests.java:111]: Could you add a test case with where and orderBy please

File: server/src/main/java/org/cloudfoundry/identity/uaa/resources/jdbc/JdbcPagingList.java (hunk 1 above)
Comment (strehle) [reply server/src/main/java/org/cloudfoundry/identity/uaa/resources/jdbc/JdbcPagingList.java]: Thanks for the hint

File: server/src/main/java/org/cloudfoundry/identity/uaa/resources/jdbc/JdbcPagingList.java [hunk 3]
```diff
@@ -73,7 +74,7 @@
         this.sql = sql;
         this.args = args;
         this.mapper = mapper;
-        this.size = parameterJdbcTemplate.queryForObject(getCountSql(sql), args, Integer.class);
+        this.size = Optional.ofNullable(parameterJdbcTemplate.queryForObject(getCountSql(sql), args, Integer.class)).orElse(0);
```
[the remark this answers is not in the document]
Comment (strehle) [reply server/src/main/java/org/cloudfoundry/identity/uaa/resources/jdbc/JdbcPagingList.java:78]: SQLs are not from users but internally

**Review by hsinn0 — APPROVED** [review]

**Pull request merged**